### PR TITLE
[TOPIC-GPIO] Samples: Bluetooth: st_ble_sensor: Update to use new GPIO API

### PR DESCRIPTION
**Pull request by mtuxpe**
Signed-off-by: Marcio Montenegro <mtuxpe@gmail.com>

### CODE REVIEW

**Comment by jhedberg** [comment]
Shouldn't topic-gpio first be rebased against master?

**Comment by mtuxpe** [comment]
Hi Johan,
Yes. You are correct.
May I cancel this PR and try another ?
Best,
Marcio

**Comment by jhedberg** [comment]
@mtuxpe of course, though I don't think another PR will fix the needed rebase

File: samples/bluetooth/st_ble_sensor/README.rst
Comment (dbkinder) [inline samples/bluetooth/st_ble_sensor/README.rst]: If you're not referencing these links in the document, it's best to delete them.

**Review by dbkinder — DISMISSED** [review]

File: samples/bluetooth/st_ble_sensor/README.rst
Comment (mtuxpe) [reply samples/bluetooth/st_ble_sensor/README.rst]: fixed

**Comment by galak** [comment]
@mtuxpe can you rebase this?

**Comment by mtuxpe** [comment]
Hi Kumar,
Got an error during rebase. I will try to solve later.
Regards,
Marcio

Applying: gpio: Update imx gpio driver to use new gpio api
Applying: drivers/gpio: stm32: Update to new GPIO api
Applying: drivers/gpio: stm32: various clean up
Applying: drivers/gpio: stm32: exti: Clear triggers when not requested
Using index info to reconstruct a base tree...
M drivers/gpio/gpio_stm32.c
A drivers/interrupt_controller/exti_stm32.c
A drivers/interrupt_controller/exti_stm32.h
Falling back to patching base and 3-way merge...
Auto-merging include/drivers/interrupt_controller/exti_stm32.h
Auto-merging drivers/interrupt_controller/intc_exti_stm32.c
Auto-merging drivers/gpio/gpio_stm32.c
Applying: boards: stm32: Update board definition using new GPIO api
Applying: tests: gpio_basic_api: Enable test on stm32 boards
Applying: boards: stm32: Fix buttons and leds configuration
Applying: boards: stm32h747i_disco: Add arduino connector header
Applying: drivers/interrupt_controller: stm32: Clean set api from unused arg
Using index info to reconstruct a base tree...
M drivers/gpio/gpio_stm32.c
A drivers/interrupt_controller/exti_stm32.c
A drivers/interrupt_controller/exti_stm32.h
Falling back to patching base and 3-way merge...
Auto-merging include/drivers/interrupt_controller/exti_stm32.h
CONFLICT (content): Merge conflict in
include/drivers/interrupt_controller/exti_stm32.h
Auto-merging drivers/interrupt_controller/intc_exti_stm32.c
Auto-merging drivers/gpio/gpio_stm32.c
error: Failed to merge in the changes.
Patch failed at 0075 drivers/interrupt_controller: stm32: Clean set api
from unused arg
hint: Use 'git am --show-current-patch' to see the failed patch

**Comment by pabigot** [comment]
Can you fix the rebase problems?  Or, with your permission, I can do the rebase for you (if my github permissions allow)?

Or, would it be ok to close this PR and I'll post https://github.com/pabigot/zephyr/commit/77d103c6fbfa428b36ed96d80e6778102aca04c3 instead?  This PR is hard to review because it seems to change things that's not directly related to the GPIO API, some of which don't seem necessary, and some of which would need to be corrected.

I understand this is short notice, but we're on a tight deadline to get the topic branch merged so unless I hear otherwise I will move to merge my conversion Monday morning.

**Comment by mtuxpe** [comment]
Hi Peter ,
Yes please you can rebase.


On Jan 26, 2020 12:43, "Peter A. Bigot" <notifications@github.com> wrote:

Can you fix the rebase problems? Or, with your permission, I can do the
rebase for you (if my github permissions allow)?

Or, would it be ok to close this PR and I'll post pabigot@77d103c
<https://github.com/pabigot/zephyr/commit/77d103c6fbfa428b36ed96d80e6778102aca04c3>
instead? This PR is hard to review because it seems to change things that's
not directly related to the GPIO API, some of which don't seem necessary,
and some of which would need to be corrected.

I understand this is short notice, but we're on a tight deadline to get the
topic branch merged so unless I hear otherwise I will move to merge my
conversion Monday morning.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/zephyrproject-rtos/zephyr/pull/21425?email_source=notifications&email_token=ABQ7USN6TVVRL5W5C7RZQQ3Q7WVSPA5CNFSM4J3MRPTKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEJ5WZ4Y#issuecomment-578514163>,
or unsubscribe

**Comment by pabigot** [comment]
Done (@carlescufi).  Note that this has not yet received any reviews.  I'll add one, but if we don't get it through to approval in the next day or so the simpler one I linked to above may have to go in, and the additional changes here done separately.

File: samples/bluetooth/st_ble_sensor/src/button_svc.c
Comment (pabigot) [inline samples/bluetooth/st_ble_sensor/src/button_svc.c]: This should not be necessary.  The devicetree source will generate that symbol.

File: samples/bluetooth/st_ble_sensor/src/button_svc.c
Comment (pabigot) [inline samples/bluetooth/st_ble_sensor/src/button_svc.c]: The interrupt should not be enabled until after the callback is configured and added.

File: samples/bluetooth/st_ble_sensor/src/led_svc.c
Comment (pabigot) [inline samples/bluetooth/st_ble_sensor/src/led_svc.c]: Remove, as above.

File: samples/bluetooth/st_ble_sensor/src/led_svc.c
Comment (pabigot) [inline samples/bluetooth/st_ble_sensor/src/led_svc.c]: Either use `LED` defined above or get rid of the local defines and use the devicetree symbol consistently.

File: samples/bluetooth/st_ble_sensor/src/main.c
Comment (pabigot) [inline samples/bluetooth/st_ble_sensor/src/main.c]: Better to log the error then `return`.  It's OK to return from `main()`.

Same for the next two uses of `ENDLESS_LOOP()`.  Then get rid of that macro.

**Review by pabigot — CHANGES_REQUESTED** [review]
A few cleanup changes needed.

**Comment by mtuxpe** [comment]
All fixed.

**Comment by carlescufi** [comment]
@mtuxpe I think you probably need a rebase, since this PR currently includes many unrelated commits.

**Comment by pabigot** [comment]
@mtuxpe I took your latest version, cherry-picked it onto the current branch, and pushed that.  You should fetch from your repository and reset your workspace to that commit 2bea8e0b15bb1f73aa1d03dd1d8a9ece0172745f so any other changes you make stay in sync.

**Review by pabigot — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Comment by pabigot** [comment]
I also just fixed the gitlint issue.  Let's let this run;  when the updated topic branch is pushed I'll rebase this one last time and we can get it merged.

File: samples/bluetooth/st_ble_sensor/src/button_svc.c
Comment (mnkp) [inline samples/bluetooth/st_ble_sensor/src/button_svc.c]: In case of a button it's better to honor pin active level, e.g. use `GPIO_INT_EDGE_TO_ACTIVE` to react on button press. This guarantees consistent behavior.

**Review by mnkp — CHANGES_REQUESTED** [review]

**Comment by mtuxpe** [comment]
I am using another notebook.Then after cloned new zephyr repository:
I can't find commit 2bea8e0.
fatal: bad object 2bea8e0b15bb1f73aa1d03dd1d8a9ece0172745f

**Comment by pabigot** [comment]
> I am using another notebook.Then after cloned new zephyr repository: I can't find commit 

You need to fetch your `msam-topic-gpio` branch from your forked repository (where you pushed the commits for this PR originally), then checkout your branch and make sure the commit in it matches the one you fetched.  If it does not you can reset it with:

    git reset --hard r.mtuxpe/msam-topic-gpio

The commit you want to modify to address review concerns is 5d70a8fcf0

File: samples/bluetooth/st_ble_sensor/src/led_svc.c
Comment (mnkp) [inline samples/bluetooth/st_ble_sensor/src/led_svc.c]: `gpio_pin_set` takes `int` as a parameter, `led_param` should be declared as `int`.

**Review by mnkp — APPROVED** [review]

**Comment by carlescufi** [comment]
@MaureenHelm I failed to see that you had requested a review from Erwan before merging, so my apologies. If @erwango can review and finds any issues we can send a separate PR

**Pull request merged**
